test/shared/src/main/scala/zio/test/TestArrow.scala

```diff
@@ -47,6 +47,13 @@ object TestResult {
 
   def any(asserts: TestResult*): TestResult = asserts.reduce(_ || _)
 
+  implicit def liftTestResultToZIO[R, E](result: TestResult)(implicit trace: Trace): ZIO[R, E, TestResult] =
+    if (result.isSuccess)
+      ZIO.succeedNow(result)
```

Comment:
I'm not 100% sure if I correctly understood the purpose of `suceedNow` here.

Reply:
It's just like `succeed` but it is private to `zio` and doesn't store the argument in a thunk. So it's faster, but you have to be careful where you use it.

Reply:
@kitlangton thanks for clarifying. What about tracing in this case? I wonder if it's okay that the implicit trace argument is ignored in the happy path.